### PR TITLE
HIPP-1172 refactor dashboard

### PR DESCRIPTION
**Pull request by rdawson-hmrc**

### CODE REVIEW

File: app/controllers/ApplicationsController.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                                 apiHubService: ApiHubService,
+                                 errorResultBuilder: ErrorResultBuilder
+                               )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport with Logging {
+
+  def onPageLoad: Action[AnyContent] = identify.async { implicit request =>
+    request.user.email.fold[Future[Result]] {
+      noEmail()
+    }(email =>
+      apiHubService.getUserApplications(email).map(userApps => Ok(view(
+        userApps.sortBy(_.name),
```
Comment (PaulCDurham) [inline app/controllers/ApplicationsController.scala:44]: Should this be case-insensitive? We've used that elsewhere. The ticket does not state this.

File: app/controllers/IndexController.scala [hunk 2]
```diff
@@ -43,14 +43,14 @@ class IndexController @Inject()(
...
     request.user.email.fold[Future[Result]] {
       noEmail()
     }(email =>
-      if (request.user.permissions.canSupport){
-          apiHubService.getApplications().map(apps => Ok(view(apps, Some(request.user))))
-      }else{
-        apiHubService.getUserApplications(email).map(userApps => Ok(view(userApps, Some(request.user))))
-      }
+      apiHubService.getUserApplications(email).map(userApps => Ok(view(
+        userApps.sortBy(_.name).take(maxApplicationsToShow),
```
Comment (PaulCDurham) [inline app/controllers/IndexController.scala:51]: Case-insensitive sort?

File: test/controllers/IndexControllerSpec.scala [hunk 3]
```diff
@@ -157,7 +132,7 @@ class IndexControllerSpec extends SpecBase with MockitoSugar with HtmlValidation
         val view = fixture.application.injector.instanceOf[IndexView]
         status(result) mustEqual OK
 
-        contentAsString(result) mustEqual view(applications, Some(FakeUser))(request, messages(fixture.application)).toString
+        contentAsString(result) mustEqual view(applications, applications.size, Some(FakeUser))(request, messages(fixture.application)).toString
         contentAsString(result) must validateAsHtml
       }
     }
```
Comment (PaulCDurham) [inline test/controllers/IndexControllerSpec.scala:138]: There's an unused import somewhere in here

**Review by PaulCDurham — COMMENTED** [review]
Do you want to check with Whitney about the case-insensitive sort?

File: app/controllers/ApplicationsController.scala (hunk 1 above)
Comment (rdawson-hmrc) [reply app/controllers/ApplicationsController.scala]: done - confirmed with Whitney that this is what we want

File: app/controllers/IndexController.scala (hunk 2 above)
Comment (rdawson-hmrc) [reply app/controllers/IndexController.scala]: done

File: test/controllers/IndexControllerSpec.scala (hunk 3 above)
Comment (rdawson-hmrc) [reply test/controllers/IndexControllerSpec.scala:138]: removed

**Review by PaulCDurham — APPROVED** [review]

**Pull request merged**